akka-actor/src/main/scala/akka/event/LoggerMailbox.scala

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.event
+
+import akka.dispatch.MessageQueue
+import akka.dispatch.MailboxType
+import akka.dispatch.UnboundedMailbox
+import com.typesafe.config.Config
+import akka.actor.ActorSystem
+import akka.actor.ActorRef
+import akka.dispatch.ProducesMessageQueue
+
+trait LoggerMessageQueueSemantics
+
+/**
+ * INTERNAL API
+ */
+private[akka] class LoggerMailboxType(settings: ActorSystem.Settings, config: Config) extends MailboxType
+  with ProducesMessageQueue[LoggerMailbox] {
+
+  override def create(owner: Option[ActorRef], system: Option[ActorSystem]) = (owner, system) match {
+    case (Some(o), Some(s)) ⇒ new LoggerMailbox(o, s)
+    case _                  ⇒ throw new IllegalArgumentException("no mailbox owner or system given")
+  }
+}
+
+/**
+ * INTERNAL API
+ */
+private[akka] class LoggerMailbox(owner: ActorRef, system: ActorSystem)
+  extends UnboundedMailbox.MessageQueue with LoggerMessageQueueSemantics {
+
+  override def cleanUp(owner: ActorRef, deadLetters: MessageQueue): Unit = {
+    if (hasMessages) {
+      var envelope = dequeue
+      while (envelope ne null) {
+        Logging.StandardOutLogger.tell(envelope.message, envelope.sender)
```

Comment:
A limit on how many we drain would be nice to have here. (`logger-max-drained-during-shutdown`?).
Would it not be a problem with the current while loop when there's someone still calling `system.log.warn(...)` and some other thread issued shutdown? Could the shutdown not happen because this actor would not let go because he'd get more and more log statements?